### PR TITLE
[SPARK-38749][SQL][TESTS] Test the error class: RENAME_SRC_PATH_NOT_FOUND

### PR DESCRIPTION
**Pull request by lvshaokang**
### What changes were proposed in this pull request?

I add a test case for the error class `RENAME_SRC_PATH_NOT_FOUND` in the QueryExecutionErrorsSuite.

### Why are the changes needed?

To improve test coverage, and record expected error messages in tests.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?
By running new test:
```
./build/sbt "test:testOnly *QueryExecutionErrorsSuite"
```

### CODE REVIEW

**Comment by lvshaokang** [comment]
@MaxGekk PTAL. Thanks.

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala [hunk 1]
```diff
@@ -616,6 +622,31 @@ class QueryCompilationErrorsSuite
         "functionName" -> "`array_contains`",
         "classCanonicalName" -> "org.apache.spark.sql.catalyst.expressions.ArrayContains"))
   }
+
+  test("RENAME_SRC_PATH_NOT_FOUND: rename the file which source path does not exist") {
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala:626]: Could you move the test from `*Compilation*Suite` to `QueryExecutionErrorsSuite`, please.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala (hunk 1 above)
Comment (lvshaokang) [reply sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala]: @MaxGekk done.

**Comment by lvshaokang** [comment]
@MaxGekk I've resolved conflicts. PTAL. Thanks!

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 2]
```diff
@@ -686,6 +689,31 @@ class QueryExecutionErrorsSuite
...
+        assert(!fm.exists(srcPath))
+        val dstPath = new Path(s"$basePath/new_file")
+        fm.renameTempFile(srcPath, dstPath, true)
+      }
+    )
+    checkError(
+      exception = e,
+      errorClass = "RENAME_SRC_PATH_NOT_FOUND",
+      parameters = Map(
+        "sourcePath" -> s"$srcPath"
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:714]: nit:
```suggestion
        "sourcePath" -> srcPath
```

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @lvshaokang.

**Pull request closed without merging**